src/utils/__tests__/env.test.ts

```diff
@@ -0,0 +1,136 @@
+          `Please use DOGS instead.`
+      );
+      expect(process.env.DOGS).toEqual('DROOL');
+    });
+
+    it('errors if neither is set', () => {
+      expect(() =>
+        checkEnvForPrerequisite({ name: 'DOGS', legacyName: 'CATS' })
+      ).toThrowError(ConfigurationError);
+    });
+  }); // end describe('with legacy name')
+
+  describe('multiple options', () => {
+    it('checks for multiple variables', () => {
+      // don't set either variable to force it to look for both (but that makes
+      // it error, so `expect` the error to catch it so it doesn't break the
+      // test)
+      expect(() =>
+        checkEnvForPrerequisite({ name: 'MAISEY' }, { name: 'CHARLIE' })
+      ).toThrowError(ConfigurationError);
+      expect(logger.debug).toHaveBeenCalledWith(
+        `Checking for environment variable(s) MAISEY or CHARLIE`
+      );
+      expect(logger.debug).toHaveBeenCalledWith(
+        `\tChecking for environment variable MAISEY`
+      );
+      expect(logger.debug).toHaveBeenCalledWith(
+        `\tChecking for environment variable CHARLIE`
+      );
+    });
+
+    it('is happy if either option is defined', () => {
+      process.env.MAISEY = 'GOOD DOG';
+      expect(() =>
+        checkEnvForPrerequisite({ name: 'MAISEY' }, { name: 'CHARLIE' })
+      ).not.toThrowError(ConfigurationError);
+      expect(logger.debug).toHaveBeenCalledWith('Found MAISEY');
+
+      delete process.env.MAISEY;
+      process.env.CHARLIE = 'ALSO GOOD DOG';
+      expect(() =>
+        checkEnvForPrerequisite({ name: 'MAISEY' }, { name: 'CHARLIE' })
+      ).not.toThrowError(ConfigurationError);
+      expect(logger.debug).toHaveBeenCalledWith('Found CHARLIE');
+    });
+
+    it('throws if neither one is defined', () => {
+      // skip defining vars here
+      expect(() =>
+        checkEnvForPrerequisite({ name: 'MAISEY' }, { name: 'CHARLIE' })
+      ).toThrowError(ConfigurationError);
+    });
+
+    it('handles a mix of variables with and without legacy names', () => {
+      process.env.MAISEY = 'GOOD DOG';
+      expect(() =>
+        checkEnvForPrerequisite(
+          { name: 'MAISEY' },
+          { name: 'CHARLIE', legacyName: 'OPAL' }
+        )
+      ).not.toThrowError(ConfigurationError);
+      expect(logger.debug).toHaveBeenCalledWith('Found MAISEY');
+
+      delete process.env.MAISEY;
+      process.env.OPAL = 'GOOD PUPPY';
+      expect(() =>
+        checkEnvForPrerequisite(
+          { name: 'MAISEY' },
+          { name: 'CHARLIE', legacyName: 'OPAL' }
+        )
+      ).not.toThrowError(ConfigurationError);
+      expect(logger.warn).toHaveBeenCalledWith(
+        `Usage of OPAL is deprecated, and will be removed in later versions. ` +
+          `Please use CHARLIE instead.`
+      );
+      expect(process.env.CHARLIE).toEqual('GOOD PUPPY');
+    });
+  }); // end describe('multiple variables')
+}); // end describe('checkEnvForPrerequisites')
```

Comment:
That's... more than an average number of tests for this project, nice :)

Reply:
Thanks! 🙂

I'll give it credit - once I realized I needed to refactor the function, the tests actually did help me catch a bug or two!